src/Handler/Table/Preview/PreviewTableHandler.php

```diff
@@ -0,0 +1,166 @@
+     * @param PreviewTableCommand $command
+     */
+    public function __invoke(
+        Message $credentials,
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof PreviewTableCommand);
+
+        // validate
+        assert($command->getPath()->count() === 1, 'CreateTableCommand.path is required and size must equal 1');
+        assert(!empty($command->getTableName()), 'CreateTableCommand.tableName is required');
+        assert($command->getColumns()->count() > 0, 'CreateTableCommand.columns is required');
+
+        try {
+            $db = $this->manager->createSession($credentials);
+            /** @var string $databaseName */
+            $databaseName = $command->getPath()[0];
+
+            // build sql
+            $columns = $this->repeatedStringToArray($command->getColumns());
+            // TODO changeSince, changeUntil
+            // TODO fulltextSearch
+            // TODO whereFilters
+            $selectTableSql = sprintf(
+                "SELECT %s %s\nFROM %s.%s",
+                $command->getLimit() ? sprintf('TOP %d', $command->getLimit()) : '',
+                implode(', ', array_map([TeradataQuote::class, 'quoteSingleIdentifier'], $columns)),
+                TeradataQuote::quoteSingleIdentifier($databaseName),
+                TeradataQuote::quoteSingleIdentifier($command->getTableName())
+            );
+
+            if ($command->hasOrderBy() && $command->getOrderBy()) {
+                $orderBy = $command->getOrderBy();
+                $selectTableSql .= sprintf(
+                    "\nORDER BY %s %s",
+                    // TODO k cemu je potreba dataType?
+                    TeradataQuote::quoteSingleIdentifier($orderBy->getColumnName()),
+                    $orderBy->getOrder() === PreviewTableCommand\PreviewTableOrderBy\Order::ASC ? 'ASC' : 'DESC'
+                );
+            }
+
+            // select table
+            $result = $db->executeQuery($selectTableSql);
+
+            // get columns definitions
+            $tableRef = new TeradataTableReflection($db, $databaseName, $command->getTableName());
+            /** @var TeradataColumn[] $columnsDefArr */
+            $columnsDefArr = iterator_to_array($tableRef->getColumnsDefinitions());
+            /** @var array<string, Teradata> $columnsDef */
+            $columnsDef = [];
+            foreach ($columnsDefArr as $index => $columnDef) {
+                $columnsDef[$columnDef->getColumnName()] = $columnDef->getColumnDefinition();
+            }
+
+            // set response
+            $response = new PreviewTableResponse();
+
+            $rows = new RepeatedField(GPBType::MESSAGE, PreviewTableResponse\Row::class);
+            foreach ($result->fetchAllAssociative() as $lineNumber => $line) {
+                // set column names
+                if ($lineNumber === 0) {
```

Comment:
nezkousel jsi fetchnout prvni radek pred foreachem nebo nastavit column z posledniho radku nebo neco tak, aby se to nemuselo resit v kazde iteraci? jakoze vykonove o nic nejde, jen mi to prijde takove zbytecne. Ale je to mini issue, klidne muze to tak zustat

Reply:
ok, upravil jsem

Reply:
tak jsem nakonec udelal jinak, pac kdyz dam napred `fetchAssociative`, tak potom `fetchAllAssociative` zacina od druheho zaznamu :-/

Nastavuju `response.columns` rovnou z `command.columns`, pac budou vzdycky stejny. A navic, kdyz se vrati prazdny vysledek, tak neni pres co iterovat, takze by to bylo prazdny.